test/integration/apiserver/admissionwebhook/mutating_webhook_gvk_conversion_test.go

```diff
@@ -0,0 +1,398 @@
+func Test_MutatingWebhookConvertsGVKWithMatchPolicyEquivalent(t *testing.T) {
+
+	roots := x509.NewCertPool()
+	if !roots.AppendCertsFromPEM(localhostCert) {
+		t.Fatal("Failed to append Cert from PEM")
+	}
+	cert, err := tls.X509KeyPair(localhostCert, localhostKey)
+	if err != nil {
+		t.Fatalf("Failed to build cert with error: %+v", err)
+	}
+
+	typeChecker := &admissionTypeChecker{}
+
+	webhookServer := httptest.NewUnstartedServer(newAdmissionTypeCheckerHandler(typeChecker))
+	webhookServer.TLS = &tls.Config{
+		RootCAs:      roots,
+		Certificates: []tls.Certificate{cert},
+	}
+	webhookServer.StartTLS()
+	defer webhookServer.Close()
+
+	upCh := typeChecker.Reset()
+	server, err := apiservertesting.StartTestServer(t, nil, []string{
+		"--disable-admission-plugins=ServiceAccount",
+	}, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+
+	etcd.CreateTestCRDs(t, apiextensionsclientset.NewForConfigOrDie(server.ClientConfig), false, versionedCustomResourceDefinition())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	config := server.ClientConfig
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Write markers to a separate namespace to avoid cross-talk
+	markerNs := "marker"
+	_, err = client.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: markerNs}}, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a marker object to use to check for the webhook configurations to be ready.
+	marker, err := client.CoreV1().Pods(markerNs).Create(context.TODO(), newMarkerPodGVKConversion(markerNs), metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	equivalent := admissionregistrationv1.Equivalent
+	ignore := admissionregistrationv1.Ignore
+
+	v1Endpoint := webhookServer.URL + "/v1"
+	markerEndpoint := webhookServer.URL + "/marker"
+	v2Endpoint := webhookServer.URL + "/v2"
+	mutatingWebhook := &admissionregistrationv1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "admission.integration.test",
+		},
+		Webhooks: []admissionregistrationv1.MutatingWebhook{
+			{
+				Name: "admission.integration.test.v2",
+				Rules: []admissionregistrationv1.RuleWithOperations{{
+					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+					Rule: admissionregistrationv1.Rule{
+						APIGroups:   []string{"awesome.bears.com"},
+						APIVersions: []string{"v2"},
+						Resources:   []string{"*/*"},
+					},
+				}},
+				MatchPolicy: &equivalent,
+				ClientConfig: admissionregistrationv1.WebhookClientConfig{
+					URL:      &v2Endpoint,
+					CABundle: localhostCert,
+				},
+				FailurePolicy:           &ignore,
+				SideEffects:             &noSideEffects,
+				AdmissionReviewVersions: []string{"v1"},
+				MatchConditions: []admissionregistrationv1.MatchCondition{
+					{
+						Name:       "test-v2",
+						Expression: "object.metadata.version == 'v2'",
```

Comment:
`object.metadata.version` isn't a field... I think this is supposed to be:

```suggestion
						Expression: "object.apiVersion == 'awesome.bears.com/v2'",
```

Reply:
yeah it'll blow up on me once I activate... sorry silly mistake

Reply:
done